celery/backends/base.py

```diff
@@ -249,7 +249,11 @@ def exception_to_python(self, exc):
                 else:
                     exc_module = from_utf8(exc_module)
                     exc_type = from_utf8(exc['exc_type'])
-                    cls = getattr(sys.modules[exc_module], exc_type)
+                    try:
+                        cls = getattr(sys.modules[exc_module], exc_type)
+                    except KeyError:
+                        cls = create_exception_cls(
+                            from_utf8(exc['exc_type']), __name__)
```

Comment:
You can use the `exc_type` variable, defined on line 251.

Reply:
Instead of `__name__` which will return `celery.backends.base` I think, you can pass `celery.exceptions.__name__`.

Reply:
1st change made.

For the second one, celery.exceptions is not imported, ok to use module name of an imported exception, e.g.

```
cls = create_exception_cls(
                            exc_type, TimeoutError.__module__)
```

Reply:
You can add the `import` statement if you don't mind, I can't find any issue with that. Your version would work too, using the module is more directly visible perhaps.